docroot/modules/custom/mass_views/src/Plugin/Action/AddCollectionsDocuments.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Drupal\mass_views\Plugin\Action;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\taxonomy\Entity\Vocabulary;
+use Drupal\views_bulk_operations\Action\ViewsBulkOperationsActionBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Allows to add Collections field value.
+ *
+ * @see https://www.drupal.org/docs/contributed-modules/views-bulk-operations-vbo/creating-a-new-action#s-2-action-class
+ *
+ * @Action(
+ *   id = "mass_views_add_documents_collections",
+ *   label = @Translation("Add Collections to Documents"),
+ *   type = "media"
+ * )
+ */
+class AddCollectionsDocuments extends ViewsBulkOperationsActionBase {
+
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($entity = NULL) {
+
+    $config = $this->getConfiguration();
+    $new_collection_id = $config['new_collection'];
+
+    $media_storage = \Drupal::entityTypeManager()->getStorage('media');
+    $vid = $media_storage->getLatestRevisionId($entity->id());
+    $create_draft = $vid != $entity->getRevisionId();
+
+    if (is_array($new_collection_id)) {
+      if (!empty($entity->field_collections->getValue())) {
+        foreach ($new_collection_id as $id) {
+          $entity->field_collections->appendItem($id);
+        }
+      }
+      else {
+        $entity->field_collections = $new_collection_id;
+      }
+    }
+
+    $entity->setNewRevision(TRUE);
+    $entity->setRevisionUserId(\Drupal::currentUser()->id());
```

Comment:
hmm, interesting. What's your suggestion?

Reply:
@deviantintegral see question above

Reply:
We could pull it out into a trait that both collection classes use? Or perhaps just a new class. This would make a good case for typed entity / bundle class logic too.

In case I wasn't clear, this could totally be a followup or a new ticket. PHPStorm will still highlight the duplicated fragments (which it doesn't now, due to injecting dependencies just here!).